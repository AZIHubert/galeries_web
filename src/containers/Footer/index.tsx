import * as React from 'react';

import Text from '#components/Text';

const Footer = () => (
  <Text
    fontWeight='lighter'
    styles={{
      fontSize: 0.65,
      marginBottom: 10,
    }}
    stylesMobile={{
      fontSize: 0.7,
    }}
    stylesLaptopL={{
      fontSize: 0.9,
    }}
  >
    Allan Aoudji @2021
  </Text>
);

export default Footer;
