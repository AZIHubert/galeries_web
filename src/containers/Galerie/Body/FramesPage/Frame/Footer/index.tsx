import moment from 'moment';
import * as React from 'react';
import { AiOutlineHeart } from 'react-icons/ai';

import Text from '#components/Text';

import theme from '#helpers/theme';

import {
  Container,
  LikeContainer,
} from './styles';

interface FooterI {
  createdAt: Date;
  likes: LikeI[];
}

const Footer = ({
  createdAt,
  likes,
}: FooterI) => (
  <Container>
    <LikeContainer>
      <Text
        fontWeight='bold'
      >
        {likes.length} like{likes.length > 1}
      </Text>
      <AiOutlineHeart
        color={theme.colors.danger}
        size={25}
      />
    </LikeContainer>
    <Text
      fontWeight='lighter'
      styles={{
        fontSize: 0.55,
        textAlign: 'center',
      }}
    >
      {moment(createdAt).calendar()}
    </Text>
  </Container>
);

export default Footer;
