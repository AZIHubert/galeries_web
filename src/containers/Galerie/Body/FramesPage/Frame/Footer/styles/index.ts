export { default as Container } from './Container';
export { default as LikeContainer } from './LikeContainer';
