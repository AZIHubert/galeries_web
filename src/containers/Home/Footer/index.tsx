import * as React from 'react';

const Footer = () => (
  <footer>
    <p>
      Allan Aoudji @2021
    </p>
  </footer>
);

export default Footer;
