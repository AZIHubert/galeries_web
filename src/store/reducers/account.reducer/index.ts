import { ACCOUNT_SET } from '#store/actions';

interface InitialStateI {
  errors: form.AccountI,
  status: store.Status,
}

const initialState: InitialStateI = {
  errors: {
    deleteAccountSentence: '',
    password: '',
    userNameOrEmail: '',
  },
  status: 'pending',
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case ACCOUNT_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
        errors: {
          ...state.errors,
          ...action.payload ? action.payload.data.errors : undefined,
        },
      };
    default:
      return state;
  }
};
