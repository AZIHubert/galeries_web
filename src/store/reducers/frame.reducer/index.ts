import { FRAME_SET } from '#store/actions';

interface InitialStateI {
  status: store.Status;
}

const initialState: InitialStateI = {
  status: 'pending',
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case FRAME_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
      };
    default:
      return state;
  }
};
