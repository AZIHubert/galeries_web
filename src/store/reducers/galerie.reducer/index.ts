import { GALERIE_SET } from '#store/actions';

interface InitialStateI {
  status: store.Status;
  errors: {
    name: string;
  }
}

const initialState: InitialStateI = {
  status: 'pending',
  errors: {
    name: '',
  },
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case GALERIE_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
        errors: {
          ...state.errors,
          ...action.payload ? action.payload.data.errors : undefined,
        },
      };
    default:
      return state;
  }
};
