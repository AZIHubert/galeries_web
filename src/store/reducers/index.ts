import { combineReducers } from 'redux';

import login from './login.reducer'; // loginError
import notification from './notification.reducer';
import sendConfirmation from './sendConfirmation.reducer'; // sendConfirmationError
import sendResetPassword from './sendResetPassword.reducer'; // sendResetPasswordError
import signin from './signin.reducer'; // SigninError
import user from './user.reducer';
import ui from './ui.reducer';

export default combineReducers({
  login,
  notification,
  sendConfirmation,
  sendResetPassword,
  signin,
  ui,
  user,
});
