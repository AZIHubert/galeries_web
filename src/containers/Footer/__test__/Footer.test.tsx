import { cleanup } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';

import ThemeProvider from '#contexts/ThemeContext';
import reducers from '#store/reducers';

import Footer from '../index';

const mockedStore = createStore(reducers);

const Container = () => (
  <Provider store={mockedStore}>
    <ThemeProvider>
      <Footer />
    </ThemeProvider>
  </Provider>
);

describe('Footer', () => {
  afterEach(cleanup);
  it('renders without crashing', () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
