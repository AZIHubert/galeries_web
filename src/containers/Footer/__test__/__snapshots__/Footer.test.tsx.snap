// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders without crashing 1`] = `
<p
  className="sc-bdfBwQ pCYUg"
  color="black"
  fontStyle="normal"
  fontWeight="lighter"
>
  Allan Aoudji @2021
</p>
`;
